### PR TITLE
Feature/nexarear

### PR DESCRIPTION
**Pull request by srippa-nexar**

### CODE REVIEW

File: scripts/eval_train.sh [hunk 1]
```diff
@@ -4,11 +4,11 @@
 # command for squeezeDet:
 # =========================================================================== #
 python ./src/eval.py \
-  --dataset=KITTI \
```
Comment (roiaddi) [inline scripts/eval_train.sh:7]: I think we shouldn't use this code at all, no? eval should happen every X steps in training

File: scripts/eval_val.sh [hunk 2]
```diff
@@ -4,11 +4,11 @@
 # command for squeezeDet:
 # =========================================================================== #
 python ./src/eval.py \
```
Comment (roiaddi) [inline scripts/eval_val.sh:6]: same here. I think we can even remove this file

File: scripts/train.sh [hunk 3]
```diff
@@ -34,11 +34,11 @@ done
 if [[ "$1" == "squeezeDet" ]]
 then
   python ./src/train.py \
-  --dataset=KITTI \
-  --pretrained_model_path=./data/SqueezeNet/squeezenet_v1.1.pkl \
-  --data_path=./data/KITTI \
+  --dataset=NEXAREAR \
```
Comment (roiaddi) [inline scripts/train.sh:37]: I think we better move to python here. 
that way we can use this code to upload the final result to s3 for example

File: src/dataset/imdb.py [hunk 4]
```diff
@@ -55,7 +55,9 @@ def data_root_path(self):
   def year(self):
     return self._year
 
-  def _shuffle_image_idx(self):
+  def _shuffle_image_idx(self,seed=None):
+    if seed is not None:
+      np.random.seed(seed=seed)
```
Comment (roiaddi) [inline src/dataset/imdb.py:60]: 👍

File: src/eval.py [hunk 5]
```diff
@@ -17,24 +17,24 @@
 import tensorflow as tf
```
Comment (roiaddi) [inline src/eval.py:17]: I don't think we need this code.

File: src/dataset/nexarear.py [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+      os.makedirs(det_file_dir)
+
+    for im_idx, index in enumerate(self._image_idx):
+      filename = os.path.join(det_file_dir, index+'.txt')
+      with open(filename, 'wt') as f:
+        for cls_idx, cls in enumerate(self._classes):
+          dets = all_boxes[cls_idx][im_idx]
+          for k in xrange(len(dets)):
+            f.write(
+                '{:s} -1 -1 0.0 {:.2f} {:.2f} {:.2f} {:.2f} 0.0 0.0 0.0 0.0 0.0 '
```
Comment (roiaddi) [inline src/dataset/nexarear.py:129]: how come? this is kitti format

File: src/dataset/nexarear.py [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+        image_dir=self._image_path,
+        image_format='.png',
+        det_error_file=det_error_file,
+        output_image_dir=det_error_dir,
+        num_det_per_type=10
+    )
+
+    return stats, ims
+
+  def analyze_detections(self, detection_file_dir, det_error_file):
```
Comment (roiaddi) [inline src/dataset/nexarear.py:188]: this code is duplicated from kitti, no need for this code duplication when using @orenmeiri benchmark code

File: scripts/train.sh (hunk 3 above)
Comment (srippa-nexar) [reply scripts/train.sh:37]: Agreed. Will change.

**Pull request merged**